### PR TITLE
[PXP-1190] Support querying array type

### PR DESCRIPTION
**Pull request by vpsx**
[PXP-1190] Support querying array type, with some caveats (see comments in code). 

Also adds new test data (new cases, in particular) and updates/fixes/adds to the tests to reflect these changes. 

### New Features
- Support querying array type.

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
Jenkins Build 18 : time taken 1 hr 0 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 19 : time taken 7 min 29 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 20 : time taken 1 min 1 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 21 : time taken 39 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 22 : time taken 47 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 23 : time taken 47 sec

File: peregrine/resources/submission/graphql/node.py [hunk 1]
```diff
@@ -217,13 +217,27 @@ def apply_arg_quicksearch(q, args, info):
...
+
     pg_props = set(getattr(q.entity(), '__pg_properties__', {}).keys())
 
     # *: filter for those with matching dictionary properties
     for key in set(args.keys()).intersection(pg_props):
         val = args[key]
-        val = val if isinstance(val, list) else [val]
-        if val:
+
+        # val is always a list, but the list elements are treated differently based on
```
Comment (fantix) [inline peregrine/resources/submission/graphql/node.py]: `val` for [`DataNode`](https://github.com/uc-cdis/peregrine/blob/c36d7ac86ef483f530055edff92992da7d17ca6c/peregrine/resources/submission/graphql/node.py#L975) is not a list but a string, then `for...in` below unpacks the string into separate characters by mistake, causing issue [PXD-2482](https://ctds-planx.atlassian.net/browse/PXP-2482). If we have no intention to change the GraphQL schema, I propose to add this harmless line back:

```python
val = val if isinstance(val, list) else [val]
```

Any thoughts pls? @vpsx @rudyardrichter

**Pull request merged**